### ISSUE TITLE
Updating indirect dependency also updates all direct dependencies

### ISSUE THREAD

**Issue by TrevorBurnham** [pnpm/pnpm#5118]
If I run `pnpm up` with a direct dependency, only that dependency is updated. But if I run `pnpm up` with an indirect dependency (supported since pnpm v6.24.4, re: https://github.com/pnpm/pnpm/issues/4122), all dependencies are updated. cc @zkochan @ai

### Code to reproduce the issue:

Create a new project with this `package.json`:

```json
{
  "name": "pnpm-update-issue",
  "private": true,
  "devDependencies": {
    "browserslist": "^4.0.0"
  }
}
```

Then run this command:

```
pnpm up caniuse-lite@latest
```

### Expected behavior

`caniuse-lite`, an indirect dependency, should be updated to the latest version without anything in `package.json` being affected.

### Actual behavior:

The version of `browserslist` specified in `package.json` is altered:

```json
    "browserslist": "^4.21.3"
```

### Additional information:

 - `node -v` prints: v16.13.0
 - Windows, macOS, or Linux?: macOS

**Comment by MattSilvaa** [comment]
I believe I am still seeing this behavior in `pnpm@8.12.1`. Is there a way to isolate `pnpm up` to only upgrade indirect dependencies @zkochan?

**Comment by foxaltus** [comment]
Yeah this still happens with latest versions. That is... huh... dangerous. I hope this one get fixed
